### PR TITLE
ST: Skip log collecting for tests which are skipped by assumptions

### PR DESCRIPTION
**Pull request by Frawless**
Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

### Type of change

- Bugfix

### Description

This PR skip log collection for tests, which are skipped/aborted by assumptions. For example tests in `watcher` package are skipped for runs where CO is installed via OLM/hem.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/main/java/io/strimzi/systemtest/logs/TestExecutionWatcher.java [hunk 1]
```diff
@@ -25,24 +26,30 @@
 
     @Override
     public void handleTestExecutionException(ExtensionContext extensionContext, Throwable throwable) throws Throwable {
-        String testClass = extensionContext.getRequiredTestClass().getName();
-        String testMethod = extensionContext.getRequiredTestMethod().getName();
-        collectLogs(testClass, testMethod);
+        if (!(throwable instanceof TestAbortedException)) {
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/logs/TestExecutionWatcher.java:29]: Just curious, is there some chance to use Junit5 method to do the same behaviour as you do here? AFAIK, `opentest4j` is used also by Junit5 this as their dependency.

**Review by see-quick — COMMENTED** [review]
Just one question. Anyway looks good.

File: systemtest/src/main/java/io/strimzi/systemtest/logs/TestExecutionWatcher.java (hunk 1 above)
Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/logs/TestExecutionWatcher.java:29]: This is what junit5 uses.

**Review by see-quick — APPROVED** [review]

**Review by scholzj — APPROVED** [review]

**Pull request merged**